spec/lib/blackout_updater_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe BlackoutUpdater do
+  describe 'update!' do
+    context 'without errors' do
+      it 'successfully updates the blackout' do
+        blackout = instance_spy(Blackout, update_attributes: nil)
+        allow(Reservation).to receive_message_chain(:overlaps_with_date_range,
+                                                    :active).and_return([])
+        updater = BlackoutUpdater.new(blackout: blackout, params: {})
+        updater.update!
+        expect(blackout).to have_received(:update_attributes)
+      end
+      it 'returns success message' do
+        blackout = instance_spy(Blackout, update_attributes: nil)
+        allow(Reservation).to receive_message_chain(:overlaps_with_date_range,
+                                                    :active).and_return([])
+        updater = BlackoutUpdater.new(blackout: blackout, params: {})
+        results = updater.update!
+        expect(results[:result]).to eq('Blackout was successfully updated.')
+      end
+      it 'does not return an error' do
+        blackout = instance_spy(Blackout, update_attributes: nil)
+        allow(Reservation).to receive_message_chain(:overlaps_with_date_range,
+                                                    :active).and_return([])
+        updater = BlackoutUpdater.new(blackout: blackout, params: {})
+        results = updater.update!
+        expect(results[:error]).to be_nil
+      end
+    end
+    context 'with errors' do
+      context 'with conflicting blackout' do
+        it 'does not update blackout' do
+          blackout = instance_spy(Blackout, update_attributes: nil)
+          cf = double('conflict', md_link: 'Test')
```

Comment:
this should be either an `instance_spy('Reservation')` or a `ReservationMock.new`, not just a plain double